### PR TITLE
Add title case style guideline

### PR DESCRIPTION
**Pull request by shannonxtreme**
Update the style guide to add guidance for title case in page titles

/sig docs
/language en
/cc @sftim

### CODE REVIEW

**Comment by bradtopol** [comment]
Assigning a SIG Docs Chair since this is trying to change the style guidelines.
/assign @natalisucks

**Comment by natalisucks** [comment]
@shannonxtreme Heya Shannon! 👋 Our style guide takes a lot of inspiration from the Google developer documentation style guide, [which recommends sentence case for titles and headings](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings). I'm wondering what the motivation is to take a different approach? If there's a conversation that yourself and Tim have had, my apologies in advance for not being aware of the context.

**Comment by shannonxtreme** [comment]
Ah @natalisucks I was trying to codify what I was seeing here: https://github.com/kubernetes/website/pull/35908#discussion_r945518552

I wasn't trying to change the guideline -- was trying to put the current convention in writing in our SG

**Comment by natalisucks** [comment]
@shannonxtreme Gotcha! It might not surprise you to hear that I don't agree with the suggested change in that comment 😂 Looking at the [docs homepage](https://kubernetes.io/docs/home/) for example, the all capitals convention isn't followed (at least mostly), and I would personally prefer consistency across the docs. However, I also see that capitalization is used across other areas of the docs very specifically (such as in our [Concepts](https://kubernetes.io/docs/concepts/) area, but not specifically in the [Contribute](https://kubernetes.io/docs/contribute/) section), so I'm now wondering what we decide on and try and ensure consistency from there.

TL;DR – We don't have a unified "current convention", but it is certainly a good idea to decide on one and stick with it.

Would be happy to hear your take @sftim if you feel strongly about this 😸

**Comment by shannonxtreme** [comment]
Fwiw I use the dev docs style guide a lot and agree with sentence case as well 😁 I think it's def helpful to have a place in our domain to point to in reviews if needed

**Comment by shannonxtreme** [comment]
Let's discuss what option we want to document in the async Slack (and thank you @bradtopol for tagging Natali!) 

/hold

File: content/en/docs/contribute/style/style-guide.md [hunk 1]
```diff
@@ -108,6 +108,15 @@ events are recorded with an associated "stage". | events are recorded with an as
...
 
+### Use title case for page titles {#title-case-titles}
+
+{{< table caption = "Do and Don't - Use title case for page titles" >}}
+Do | Don't
+:--| :-----
+Kubernetes API Server Bypass Risks | Kubernetes API server bypass risks
+Configure a Security Context for a Pod or Container | Configure a security context for a Pod or container
+{{< /table >}}
+
```
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:119]: Can we clarify that this is just for the title in the front matter, and not for headings within the page?

**Comment by sftim** [comment]
@natalisucks to me this isn't a _change_; it's codifying the existing convention (which is not that well followed or as well known).

I learned that convention from approvers / reviewers telling me about it.

**Comment by sftim** [comment]
I'd kind of like to merge this PR so it becomes part of the style guide history, even if we then decide we want to change from that convention.

That'll help someone looking for a answer to a question like “Hmm, when did we adopt the _previous_ convention and what was our thinking when we did”.

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (a-mccarthy) [reply content/en/docs/contribute/style/style-guide.md:119]: Should this convention apply to all titles, page and heading? I dont think that we have a guideline for headings (i couldn't see it in the style guide, please point it out to me if its there already) but it seems to be the general convention that we use in headings as well

Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md:119]: We [write headings (not titles) in sentence case](https://kubernetes.io/docs/contribute/style/style-guide/#headings).

Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md:119]: The guidance on page titles could link to the advice on headings, maybe also vice versa?

Comment (shannonxtreme) [reply content/en/docs/contribute/style/style-guide.md:119]: Might be a good idea to merge the titles section into the heading section, call it "Headings and titles" maybe, and clarify what works where. 

And I'd still like us to discuss changing the guidance for titles to use sentence case as well, but perhaps as a separate issue. The 5 votes that got in Slack arent many though

**Comment by reylejano** [comment]
I agree that we should document that title case is used for the front matter title
We should be clear to differentiate front matter title vs headings
Maybe prepend the example with `title: `
or have a larger example like this from the configmaps concepts page

> ---
> title: ConfigMaps
> content_type: concept
> weight: 20
> ---

**Comment by sftim** [comment]
The title of the ConfigMaps page is, unfortunately, both sentence case _and_ title case (due to our guidelines about how to capitalize API kinds). Maybe a different example?

**Comment by shannonxtreme** [comment]
@sftim and @reylejano I merged this into the headings section which I renamed to Headings and titles (added an anchor to not break links). I prepended the example with `title:` to make it clear we mean the frontmatter.

File: content/en/docs/contribute/style/style-guide.md [hunk 2]
```diff
@@ -459,7 +459,13 @@ Do | Don't
 Update the title in the front matter of the page or blog post. | Use first level heading, as Hugo automatically converts the title in the front matter of the page into a first-level heading.
 Use ordered headings to provide a meaningful high-level outline of your content. | Use headings level 4 through 6, unless it is absolutely necessary. If your content is that detailed, it may need to be broken into separate articles.
 Use pound or hash signs (`#`) for non-blog post content. | Use underlines (`---` or `===`) to designate first-level headings.
-Use sentence case for headings. For example, **Extend kubectl with plugins** | Use title case for headings. For example, **Extend Kubectl With Plugins**
+Use sentence case for headings in the page body. For example, **Extend kubectl
+with plugins** | Use title case for headings in the page body. For example,
```
Comment (chrismetz09) [inline content/en/docs/contribute/style/style-guide.md:463]: Bold tags appearing as text in preview.

**Comment by chrismetz09** [comment]
> Contribute
FWIW, I followed the google dev style guide when I was working on the contribute section - Use sentence case for all headings.

**Comment by shannonxtreme** [comment]
> > Contribute
> FWIW, I followed the google dev style guide when I was working on the contribute section - Use sentence case for all headings. 
> 

Yeah I will open a new issue that we can use to implement that as official guidance for both headings and titles!

**Comment by natalisucks** [comment]
Thanks for following up with this, @shannonxtreme! Given the feedback throughout the PR, I'm happy to sign-off on this being added to make our style guide clearer.
/lgtm

File: content/en/docs/contribute/style/style-guide.md [hunk 3]
```diff
@@ -459,7 +459,13 @@ Do | Don't
...
+**Extend Kubectl With Plugins**
+Use title case for the page title in the frontmatter. For example,
```
Comment (chrisnegus) [inline content/en/docs/contribute/style/style-guide.md:465]:
```suggestion
Use title case for the page title in the front matter. For example,
```
We refer to front matter as two words in other places in this doc.

File: content/en/docs/contribute/style/style-guide.md [hunk 4]
```diff
@@ -459,7 +459,13 @@ Do | Don't
...
+`title: Kubernetes API Server Bypass Risks` | Use sentence case for page titles
+in the frontmatter. For example, don't use
```
Comment (chrisnegus) [inline content/en/docs/contribute/style/style-guide.md:467]:
```suggestion
in the front matter. For example, don't use
```

**Review by chrisnegus — COMMENTED** [review]
Small wording suggestion.

**Comment by sftim** [comment]
> I'd kind of like to merge this PR so it becomes part of the style guide history, even if we then decide we want to change from that convention.
> 
> That'll help someone looking for a answer to a question like “Hmm, when did we adopt the _previous_ convention and what was our thinking when we did”.

/approve

We can track a new(?) issue to change this guidance
